scripts/build/termux_step_massage.sh

```diff
@@ -82,7 +82,8 @@ termux_step_massage() {
 			if head -c 100 "$file" | head -n 1 | grep -E "^#!.*/bin/.*" | grep -q -E -v -e "^#! ?/system" -e "^#! ?$TERMUX_PREFIX_CLASSICAL"; then
 				sed --follow-symlinks -i -E "1 s@^#\!(.*)/bin/(.*)@#\!$TERMUX_PREFIX/bin/\2@" "$file"
 			fi
-		done < <(find -L . -type f -print0)
+		# Find files with '#!' signature in the first two bytes (null-delimited)
+		done < <(find . -type f -print0 | perl -0ne 's/\0$//; open F,"<",$_;read(F,$b,2);print "$_\0" if $b eq "#!";')
```

Comment:
*Why are we embedding a perl one-liner now?*
Readability definitely isn't the reason.

Reply:
They let us scan first bytes of file instead of scanning the whole file (like grep) which will work much better for big files.

Reply:
`head -c4 "$FILE" | grep -qP`

Reply:
https://github.com/termux/termux-packages/pull/24518#discussion_r2062812869